include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -485,23 +479,21 @@ struct reduce_over_sub_group
  
     template <typename _NDItemId, typename _Size, typename _AccLocal>
     inline _Tp
-    reduce_impl(const _NDItemId& __item_id, const _Size& __n, const _AccLocal& __local_mem,
+    reduce_impl(const _NDItemId& __item_id, const _Size& __n, const _Tp& __val, const _AccLocal& __local_mem,
                 std::false_type /*has_known_identity*/) const
     {
       auto __sg = __item_id.get_sub_group();
 
       auto __local_idx = __item_id.get_local_id(0);
 
-      _Tp __val = __local_mem[__local_idx];
-
       return shuffle_sub_group_reduce(__val, __sg, __n, std::false_type());
     }
 
     template <typename _NDItemId, typename _Size, typename _AccLocal>
     inline _Tp
-    operator()(const _NDItemId& __item_id, const _Size& __n, const _AccLocal& __local_mem) const
+    operator()(const _NDItemId& __item_id, const _Size& __n, const _Tp& __val, const _AccLocal& __local_mem) const
     {
-        return reduce_impl(__item_id, __n, __local_mem, __has_known_identity<_BinaryOperation1, _Tp>{});
+        return reduce_impl(__item_id, __n, __val, __local_mem, __has_known_identity<_BinaryOperation1, _Tp>{});
     }
 
     template <typename _InitType, typename _Result>
```

Comment:
I think `reduce_over_sub_group::local_mem_req` should return 0, right?

Reply:
Done